### DISCUSSION TITLE
❓ How to rename saved wav files upon saving?

### DISCUSSION THREAD

**Discussion by ProjCRys** [snakers4/silero-models#262] [Q&A] [answered]
## ❓ Questions and Help

We have a [wiki](https://github.com/snakers4/silero-models/wiki) available for our users. Please make sure you have checked it out first.

Okay when I try to run this example it always create a 'test.wav' file. I wanted to have a custom name or file path where it could be saved because I'm working on a project that uses this through multithreading. The 'test.wav' file can only exist once in a folder. There is also no mention of renaming the saved wav file in the documentation, (unless I missed it and please tell me where).

# V3 EN
import os
import torch

device = torch.device('cpu')
torch.set_num_threads(4)
local_file = 'model.pt'

if not os.path.isfile(local_file):
    torch.hub.download_url_to_file('https://models.silero.ai/models/tts/en/v3_en.pt',
                                   local_file)  

model = torch.package.PackageImporter(local_file).load_pickle("tts_models", "model")
model.to(device)

example_text = 'Hello there.'
sample_rate = 48000
speaker='en_11'

audio_paths = model.save_wav(text=example_text,
                             speaker=speaker,
                             sample_rate=sample_rate)

![image](https://github.com/snakers4/silero-models/assets/127005727/22dffeef-e128-45d0-8444-845c0cf7ddb6)

**Comment by snakers4** [comment, marked as answer]
The `model` has `model.save_wav` method that has the same params as apply tts, but also has an `audio_path` parameter.

But I encourage you to use the codec of your liking and save the audio by yourself. Beware that the model may output float values and some codecs / libraries may not check the inputs or require int values.

**Reply by ProjCRys** [reply]
Thank you for the answer. I'll leave the code with your suggestion for others to easily copy paste here:

- - - - -

# Updated script with custom file path and name for saved audio

import os
import torch

device = torch.device('cpu')
torch.set_num_threads(4)
local_file = 'model.pt'

if not os.path.isfile(local_file):
    torch.hub.download_url_to_file('https://models.silero.ai/models/tts/en/v3_en.pt', local_file)

model = torch.package.PackageImporter(local_file).load_pickle("tts_models", "model")
model.to(device)

example_text = 'Hello there.'
sample_rate = 48000
speaker = 'en_11'

# Update: Save the audio with a custom file path and name
# Use the model.save_wav method with the audio_path parameter
custom_audio_path = r'path\to\save\custom_name.wav'
audio_paths = model.save_wav(
    text=example_text,
    speaker=speaker,
    sample_rate=sample_rate,
    audio_path=custom_audio_path
)
